### PR TITLE
Implement connectStart in PerformanceResourceTiming

### LINKED ISSUE #21261: PerformanceResourceTiming: connectStart
If a persistent connection is used, connect_start will be set in components/net/http_loader.rs::http_fetch, otherwise, it will be in components/net/http_loader.rs::obtain_response

On getting, the connectStart attribute MUST return as follows:

1. The same value as fetchStart, if a persistent connection [RFC7230] is used or the resource is retrieved from relevant application caches or local resources.
2. The time immediately before the user agent start establishing the connection to the server to retrieve the resource, if the last non-redirected fetch of the resource passes the timing allow check algorithm. If the transport connection fails and the user agent reopens a connection, connectStart SHOULD return the corresponding value of the new connection.
3. zero, otherwise.

https://w3c.github.io/resource-timing/#dom-performanceresourcetiming-connectstart

### PR DESCRIPTION
**Pull request by nehalem501**
- Added connectStart where needed (```ResourceFetchTiming```, ```ResourceAttribute```) in ```components/net_traits/lib.rs ```
- Before calling ```client.request``` in ```obtain_response```, we now add the ```connectStart``` attribute (```components/net/http_loader.rs```)
- Updated tests to reflect those changes

---

- [X] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes fix #21261 (at least partially)

- [X] There are tests for these changes OR
- [ ] These changes do not require tests because ___

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23323)

### CODE REVIEW

**Comment by jdm** [comment]
@bors-servo r+
Thanks!

**Comment by jdm** [comment]
Ok, the error is triggered by #21350, which will be fixed by #23322.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini [hunk 1]
```diff
@@ -97,9 +91,6 @@
   [PerformanceResourceTiming interface: resource must inherit property "domainLookupStart" with the proper type]
     expected: FAIL
 
-  [PerformanceResourceTiming interface: resource must inherit property "connectStart" with the proper type]
```
Comment (jdm) [inline tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Let's revert this change and the next one in this file. That should allow us to merge this PR.

**Review by jdm — CHANGES_REQUESTED** [review]

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini (hunk 1 above)
Comment (nehalem501) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: I've pushed the revert to fix this.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini [hunk 2]
```diff
@@ -148,6 +151,9 @@
   [PerformanceResourceTiming interface: attribute domainLookupStart]
     expected: FAIL
 
+  [PerformanceResourceTiming interface: attribute connectStart]
+    expected: FAIL
```
[the remark this answers is not in the document]
Comment (nehalem501) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Should be ok now.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini [hunk 3]
```diff
@@ -44,12 +44,18 @@
...
+
   [PerformanceResourceTiming interface: attribute workerStart]
     expected: FAIL
 
   [PerformanceResourceTiming interface: attribute domainLookupStart]
     expected: FAIL
 
+  [PerformanceResourceTiming interface: attribute connectStart]
+    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Please remove this particular test result, since it passes with your changes.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini [hunk 4]
```diff
@@ -44,12 +44,18 @@
   [PerformanceResourceTiming interface: resource must inherit property "domainLookupStart" with the proper type]
     expected: FAIL
 
+  [PerformanceResourceTiming interface: resource must inherit property "connectStart" with the proper type]
+    expected: FAIL
```
Comment (jdm) [inline tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Please remove this particular test result, since it passes with your changes.

**Review by jdm — CHANGES_REQUESTED** [review]
It looks like my previous comments have not been precise enough. Your original changes were fine, with the exception of one particular test result which needed to be removed.

File: tests/wpt/metadata/resource-timing/idlharness.any.js.ini (hunk 4 above)
Comment (nehalem501) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Ok, sorry for the mistake.
So if I understood it well, this is the changes I need to make:
```
diff --git a/tests/wpt/metadata/resource-timing/idlharness.any.js.ini b/tests/wpt/metadata/resource-timing/idlharness.any.js.ini
index 46f3c03ddc..1938b2ca3f 100644
--- a/tests/wpt/metadata/resource-timing/idlharness.any.js.ini
+++ b/tests/wpt/metadata/resource-timing/idlharness.any.js.ini
@@ -44,18 +44,12 @@
   [PerformanceResourceTiming interface: resource must inherit property "domainLookupStart" with the proper type]
     expected: FAIL
 
-  [PerformanceResourceTiming interface: resource must inherit property "connectStart" with the proper type]
-    expected: FAIL
-
   [PerformanceResourceTiming interface: attribute workerStart]
     expected: FAIL
 
   [PerformanceResourceTiming interface: attribute domainLookupStart]
     expected: FAIL
 
-  [PerformanceResourceTiming interface: attribute connectStart]
-    expected: FAIL
-
   [PerformanceResourceTiming interface: resource must inherit property "redirectEnd" with the proper type]
     expected: FAIL
```

Comment (jdm) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Correct.

Comment (nehalem501) [reply tests/wpt/metadata/resource-timing/idlharness.any.js.ini]: Should be good now.

**Comment by jdm** [comment]
@bors-servo r+
Sorry about the delay!

**Comment by atouchet** [comment]
Is #21261 finished or is there still more to do for that?
